lighthouse-core/report/templates/report-template.html

```diff
@@ -177,3 +176,13 @@ <h1 class="menu__header-title">Lighthouse</h1>
     </footer>
   </section>
 </div>
+
+{{#if scripts }}
+<script>
+  window.lhresults = {{{lhresults}}};
```

Comment:
> The thing we're trying to prevent is a malicious actor creating gists in a user's account

Ideally we should prevent them from doing anything, of course. It's better that way but it also helps us down the line as we add new features, things get more complicated, etc. `lhresults` is still used in the extension's report, and while it's stripped out in `replaceReportHTML()` right now, it's possible we'll want to use it at some point in the future.

> I'm not sure this does anything different other than adding an additional parse.

It is an additional parse, but it means the only cost is an additional parse. Meanwhile everything is escaped and no matter the user generated content it won't be able to break out. If we assign the object in js we're doing an eval with an optional innerHTML if it can close the element, with all the danger that entails. JS escaping is fairly complicated; this seems like an easy way to make sure we have it right (at least as a defense-in-depth thing with the rest of our efforts).

> Handlebars, #1188, and #996 already force entity escaping everywhere that's needed.

The various other template entries are protected, but `{{{lhresults}}}` is raw and only protected by escaping the closing `</script>` tag from #996. Unless we're very careful it's possible someone will be able to do the double escape trick to get those back (or try one of the various other tricks in that doc) with our current system or in the future. Handlebars and the markdown change don't touch `lhresults` since it's using the triple stash, and we're adding new user content all the time (e.g. the new CSS usage audit, which adds verbatim quotes from stylesheets in its `extendedInfo`).

> Given the current low volume of sharers, let's land this so we can grow some us LH users!

Totally agree. The main thing is that this is a fairly trivial change with no real overhead (semantically or performance-wise). The escaping that needs to happen is small in scope and already handled by Handlebars. It keeps us to the guarantee that the only things in `{{{}}}` is content completely generated by LH itself.

It's really just
```
<div id="lhresults">{{lhresults}}</div>
{{#if scripts }}
<script>
  // Use escaped results object to protect against XSS from result details.
  window.lhresults = JSON.parse(document.querySelector('#lhresults').textContent);
</script>
{{/if}}
```
(with `display: none` in the css) and everything else acts as it does now with no other changes.